src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTests.cs

```diff
@@ -13292,6 +13292,96 @@ .maxstack  4
 ");
         }
 
+        [WorkItem(24348, "https://github.com/dotnet/roslyn/issues/24348")]
+        [Fact]
+        public void VarargBridgeSource()
+        {
+
+            var code = @"
+public class VarArgs
+{
+    public void Invoke(__arglist)
+    {
+        var ai = new System.ArgIterator(__arglist);
+        System.Console.WriteLine(ai.GetRemainingCount());
+    }
+}
+
+interface IVarArgs
+{
+    void Invoke(__arglist);
+}
+
+class MyVarArgs : VarArgs, IVarArgs
+{
+
+}
+
+public static class P
+{
+    public static void Main()
+    {
+        IVarArgs iv = new MyVarArgs();
+
+        iv.Invoke(__arglist(1, 2, 3, 4));
+    }
+}
+";
+
+            var compilation = CompileAndVerify(code, expectedOutput: "4");
+        }
+
+        [WorkItem(24348, "https://github.com/dotnet/roslyn/issues/24348")]
+        [Fact]
+        public void VarargBridgeMeta()
+        {
+            var reference = CreateStandardCompilation(@"
+public class VarArgs
+{
+    public void Invoke(__arglist)
+    {
+        var ai = new System.ArgIterator(__arglist);
```

Comment:
Nit: Method body could be removed (`throw null` instead)

Reply:
That is from the original repro. I think it is ok to have. At least as some extra coverage.

---
In reply to: [166780416](https://github.com/dotnet/roslyn/pull/24679#discussion_r166780416) [](ancestors = 166780416)